tests/test_shuffle.py

```diff
@@ -1,12 +1,70 @@
+import uuid
+
 import dask
+import dask.dataframe as dd
+import pytest
+
+
+@pytest.fixture(scope="session")
+def s3_stability_url(s3, s3_bucket):
+    # Ensure that the stability-scratch directory exists,
+    # but do NOT reomove it as multiple test runs could be
+    # accessing it at the same time
+    stability_url = f"{s3_bucket}/stability-scratch"
+    s3.mkdirs(stability_url, exist_ok=True)
+    return stability_url
+
+
+@pytest.fixture(scope="session")
+def s3_test_data_url(s3, s3_stability_url):
+    # Unique, because multiple tests are accessing the bucket
+    test_url = f"{s3_stability_url}/test-data-{uuid.uuid4().hex}"
+
+    try:
+        s3.makedirs(test_url)
+        yield test_url
+    finally:
+        s3.rm(test_url, recursive=True)
+
 
+@pytest.fixture
+def s3_write_url(s3, s3_stability_url):
+    # Unique, because multiple tests are accessing the bucket
+    write_url = f"{s3_stability_url}/test-output-{uuid.uuid4().hex}"
 
-def test_shuffle_simple(small_client):
+    try:
+        s3.makedirs(write_url)
+        yield write_url
+    finally:
+        s3.rm(write_url, recursive=True)
+
+
+@pytest.fixture
+def shuffle_dataset(small_client, s3_test_data_url, s3_storage_options):
+    """Produces a ~80GB dataset which is about the memory limit of the cluster"""
     df = dask.datasets.timeseries(
-        start="2000-01-01", end="2000-12-31", freq="1s", partition_freq="1D"
+        start="2000-01-01", end="2000-12-31", freq="50ms", partition_freq="1D"
     )
 
-    sdf = df.shuffle(on="x")
+    df.to_parquet(
+        f"s3://{s3_test_data_url}",
+        compute=True,
+        overwrite=True,
+        storage_options=s3_storage_options,
+    )
 
-    sdf_size = sdf.memory_usage(index=True).sum() / (1024.0**3)
-    assert sdf_size.compute() > 1.0  # 1.0 GB
+    yield dd.read_parquet(
+        f"s3://{s3_test_data_url}", storage_options=s3_storage_options
+    )
+
+
+@pytest.mark.stability
+def test_shuffle_simple(shuffle_dataset, s3_storage_options, s3_write_url):
+    sdf = shuffle_dataset.shuffle(on="x")
+    write = sdf.to_parquet(
+        f"s3://{s3_write_url}",
+        compute=False,
+        overwrite=True,
```

Comment:
Are these options needed? I would have expected the default values for these parameters to work

Reply:
I've removed them and used the defaults

Reply:
So I found something interesting. when I omit `compute=True` and `overwrite=False` there's a lot more spilling to disk and the tests can take up to an hour!

I'm wondering if `compute=True` results in a persist somehow in the distributed case somehow.